public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx

```diff
@@ -70,6 +70,11 @@ export const LokiQueryEditor = React.memo<LokiQueryEditorProps>((props) => {
           setParseModalOpen(true);
           return;
         }
+
+        if (!result.query.labels.length && !result.query.operations.length && query.expr.length > 0) {
+          setParseModalOpen(true);
+          return;
+        }
```

Comment:
For the nodes that we don't support in query builder, we usually juts add error in https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/querybuilder/parsing.ts#L118C17-L118C33. We have a utility function `createNotSupportedError` specifically for this. You can use https://github.com/grafana/lezer-logql/blob/main/tools/tree-viz.html to check what are the nodes and in this case, I don't think we don't support in query builder any `LiteralExpr` if it is not in `BinaryExpr`.  So I would add that check there, to have a single source of truth where we check for supported/not supported nodes. 

What do you think?  
<img width="698" alt="image" src="https://github.com/grafana/grafana/assets/30407135/855b22eb-b890-4ec9-899b-841755fcc3fa">
<img width="1015" alt="image" src="https://github.com/grafana/grafana/assets/30407135/bc29fbdd-47e9-479d-93ca-d81a7406f53a">

Reply:
Actually it looks like the query builder doesn't currently handle literals within a binary expression either:

<img width="465" alt="image" src="https://github.com/grafana/grafana/assets/109082771/0e188608-860e-4517-a097-bb7ea95d4f08">
<img width="574" alt="image" src="https://github.com/grafana/grafana/assets/109082771/70675a95-0903-4ef1-b368-583a03c65e69">
<img width="504" alt="image" src="https://github.com/grafana/grafana/assets/109082771/0be16e5f-9b09-4391-901f-fd759d808b84">
<img width="659" alt="image" src="https://github.com/grafana/grafana/assets/109082771/ae7e887f-a555-48df-bb94-c944f864e243">

Maybe we want to show the warning modal for any expressions as part of this PR, and then follow up with another issue/PR to address converting literal expressions properly?

Reply:
Yes, make sense!

Reply:
Also it looks like we'll need to change the logic around resetting errors for empty queries if we want the error node to drive this UI. https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/querybuilder/parsing.ts#L112-L114, we're purposefully stripping out any errors when the query is empty

Reply:
I tweaked this PR to re-use the `isQueryEmpty` (renamed `isLokiQueryEmpty`) instead of duplicating the logic in LokiQueryEditor as I was before, the logic should be the same, but it's a bit DRYer now.

I'll open up another issue and start investigating the broader issue here, but I think this PR is fine to be released on its own as is?

Reply:
I noticed that we're doing two different things in two places. We remove errors for empty queries intentionally, but we also do the same check and then display the error modal. As this is not a critical/big bug, I'd think that we can rather solve this properly within 1 PR.

Reply:
Cool I'll close this out and try to address the root problem in https://github.com/grafana/grafana/issues/74939, and https://github.com/grafana/grafana/pull/74950/files